functions/src/timetables/Timetable.controller.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Path,
+  Post,
+  Response,
+  Route,
+  SuccessResponse,
+} from 'tsoa';
+import { Timetable } from './Timetable.model';
+import {
+  addTimetable,
+  getTimetable,
+  TimetableParams,
+  TimetableReturn,
+} from './Timetable.service';
+
+interface ValidateErrorJSON {
+  message: 'Validation failed';
+  details: { [name: string]: unknown };
+}
+
+@Route('timetables')
+export class TimetablesController extends Controller {
+  @Response(404, 'Not found')
+  @Get('{slug}')
+  public async getTimetable(@Path() slug: string): Promise<Timetable | void> {
+    // set the Cache-Control for 24h.
+    this.setHeader(
+      'Cache-Control',
+      `public, max-age=${3600}, s-max-age=${3600}, stale-while-revalidate=${30}, stale-if-error=${60}`
+    );
```

Comment:
This can be cached longer as generally we don't change the timetable data.

Reply:
what do you think would make sense? I could do a week or something

Reply:
Keep it this for now I think since it's the first release :shrug: